SmileAuth/Classes/SmileAuthenticator.m

```diff
@@ -304,4 +304,7 @@ +(void)clearPassword{
     [[SmileAuthenticator sharedInstance].keychainWrapper writeToKeychain];
 }
 
+-(void)changeAuthentication:(BOOL)newAuth {
+    _isAuthenticated = newAuth;
+}
 @end
```

Comment:
Could you tell me where & when this method is called?  In `ViewDidAppear` or other case?

Because  `SmileAuthenticator` observe `UIApplicationDidEnterBackgroundNotification`, add call below method:

```Objective-c
-(void)appDidEnterBackground:(NSNotification*)notification{
    if (_isAuthenticated) {
        _didReturnFromBackground = YES;
        _isAuthenticated = NO;
    }
}
```
so even manually change `_isAuthenticated`, this property will be changed when app did enter background.